arbnode/seq_coordinator.go

```diff
@@ -559,6 +571,7 @@ func (c *SeqCoordinator) update(ctx context.Context) time.Duration {
 			log.Error("myurl main sequencer, but no sequencer exists")
 			return c.noRedisError()
 		}
+		c.sequencer.Pause()
 		err := c.chosenOneUpdate(ctx, localMsgCount, localMsgCount, nil)
```

Comment:
Is there concern of a race here, if a transaction first reaches the sequencer before it's paused, but only reaches the transaction streamer after `chosenOneUpdate` has marked it as having the lockout, therefore getting sequenced before delayed messages?

Reply:
Note that to be here we can't be the previous chosen - otherwise we'd have gone to updatePrevKnownChosen in start of this function.
When reaching this line the sequencer is already either paused or forwarding, and not sequencing - so a tx running in parallel will either be forwarded or paused, either is fine.
Agree it's not clear. I'll add a comment for that. LMK if it makes it clearer.

There could theoretically be some weird edge case where we somehow lost and regained the lockdown very quickly. In this case - it doesn't really matter if we scheduled the tx in parallel before or after the delayed tx as there isn't really a need to schedule the delayed tx first.

Reply:
Ah, I see. That makes sense.